src/DSharp.Compiler/Compiler/MetadataBuilder.cs

```diff
@@ -255,9 +266,24 @@ public MetadataBuilder(IErrorHandler errorHandler)
             return types;
         }
 
+        private IEnumerable<(TypeSymbol, IEnumerable<MethodSymbol>)> FetchTypesWithExtensionMethods(IEnumerable<TypeSymbol> typeSymbols)
+        {
+            return typeSymbols.Where(symbol => symbol.IsPublic || symbol.IsInternal).Select(type =>
+            {
+                return (type, type.Members.Where(m => IsExtensionMethod(m)).Cast<MethodSymbol>());
+            });
+        }
+
+        private static bool IsExtensionMethod(MemberSymbol memberSymbol)
+        {
+            return memberSymbol is MethodSymbol methodSymbol 
+                && methodSymbol.IsExensionMethod 
+                && (memberSymbol.Visibility.HasFlag(MemberVisibility.Public) || memberSymbol.IsInternal);
+        }
+
         private void BuildAssembly(ParseNodeList compilationUnits)
         {
-            string scriptName = GetAssemblyScriptName(compilationUnits);
+            string scriptName = GetAssemblyScriptName(compilationUnits) ?? "g_" + Guid.NewGuid().ToString().Replace("-", "");
```

Comment:
what

Reply:
Oh this is a hack to always build without an assembly name! Forgot about this